packages/itwinui-react/src/utils/functions/react.ts

```diff
@@ -23,8 +23,13 @@ export const cloneElementWithRef = (
     return children;
   }
 
-  // Supporting React 19 and 18
-  const childrenRef = children.props?.ref || (children as any)?.ref;
+  // Supporting React 19 and earlier versions
+  const childrenRef = (() => {
+    if (React.version.startsWith('19.')) {
```

Comment:
`React.version` does not look like a documented API. I think we should avoid using it to make assumptions about the future.

Instead, we can reverse the approach:
1. Make a utility function `isReact17or18()`.
2. Use `element.ref` if the condition matches.

Reply:
I'm not suggesting avoiding `React.version` altogether, I'm only suggesting not relying on it being available in the _future_. We know for a fact that it exists in previous versions (which will no longer be updated).

So something like this should work:

```js
import * as React from "react";

const _React = React;

function isReact17or18 {
  const version = _React.version?.split(".")?.[0];
  return ["17", "18"].includes(version);
}
```

The `_React` reassignment helps bypass bundler optimizations, and the `?` operator helps at runtime.